chall04/edramire.py

```diff
@@ -0,0 +1,52 @@
+#! /usr/bin/env python3
+
+import sys
+import fileinput
+
+
+def process_file(filename):
+    shelfs = []
+    books = 0
+    try:
+        for line in fileinput.input(filename):
+            shelfs_str = line.split(" ")
+            if len(shelfs) == 0:
+                for word in shelfs_str:
+                    word = word.rstrip('\n\r')
+                    if len(word) == 0:
+                        continue
+                    if not word.isdigit():
+                        print(f"./edramire.py: {filename or 'stdin'}: Can't read file")
+                        return
+                    shelfs.append(int(word))
+            else:
+                if len(shelfs_str) <= 1 or not shelfs_str[0].isdigit():
+                    print(f"./edramire.py: {filename or 'stdin'}: Can't read file")
+                    return
+                books += int(shelfs_str[0])
+        shelfs.sort(reverse=True)
+        for k in range(len(shelfs)):
+            books -= shelfs[k]
+            if books <= 0:
+                print(k + 1)
+                return
```

Comment:
Have you tried an input like:
```
10 10
20 book1
```
You can't split a book in to, this input is impossible, however your code returns that I need 2 shelves., however this case should return an error, a none of the shelves fits your book.